### ISSUE TITLE
cmd/compile: checkbce failing in 'noopt' builder

### ISSUE THREAD

**Issue by bcmills** [golang/go#27684]
Example: https://build.golang.org/log/15361a99186b257b1df1e9f95fa505b3ffcc1a66

```
##### ../test
# go run run.go -- checkbce.go

checkbce.go:164: missing error "Found IsInBounds$"
checkbce.go:166: missing error "Found IsInBounds$"
checkbce.go:168: missing error "Found IsInBounds$"
Unmatched Errors:
checkbce.go:165: Found IsSliceInBounds
checkbce.go:167: Found IsSliceInBounds
checkbce.go:169: Found IsSliceInBounds

FAIL	checkbce.go	0.061s
```

CC: @randall77 @josharian @martisch

**Comment by randall77** [comment]
I'm confused by this.  Shouldn't we not be testing bounds check elimination on the noopt builder?
The prove pass certainly isn't running on this builder.
There may be cases where the noopt builder removes a bounds check, but it certainly won't be as aggressive as the standard builder with all the bounds check machinery enabled.

**Comment by josharian** [comment]
See also discussion at https://go-review.googlesource.com/107355 which is when this started failing.

**Comment by randall77** [comment]
I guess somehow `GO_GCFLAGS` isn't making to the recursive calls to `go tool compile` that the test harness does.
Which may be a good thing, otherwise lots of things in `test` would fail.
But that leaves the conundrum of why this test is failing on the noopt builder.

**Comment by randall77** [comment]
This was fixed when #27833 was fixed.
